libraries/lib-ipc/IPCServer.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <memory>
+
+class IPCChannel;
+class IPCChannelStatusCallback;
+
+/**
+ * \brief Simple TCP socket based ipc server.
+ * Currently, implementation allows only one channel per
+ * instance. IPCChannelStatusCallback methods are always called
+ * from the worker thread, though IPCChannel instance created
+ * by this implementation is thread safe (considering lifetime limitations)
+ */
+class IPC_API IPCServer final
+{
+   class Impl;
+   std::unique_ptr<Impl> mImpl;
+public:
+
+   IPCServer();
+   ~IPCServer();
+
+   /**
+    * \brief Creates channel and starts listening for the incoming connection.
+    * \param id Basically is a tcp port number used to establish a connection
+    * \param callback In case of successful connection callback should exist
+    * until IPCChannelStatusCallback::OnDisconnect or
+    * IPCChannelStatusCallback::OnConnectionError called or server instance
+    * destroyed.
+    * \return true if channel was successfully created and waiting for incoming
+    * connection
+    */
+   bool CreateChannel(int id, IPCChannelStatusCallback* callback);
```

Comment:
- callback is assumed non-null inside the function -- therefore, make it a reference argument
- id is a "naked" int.  I would write a small ChannelId class wrapping the int and preventing implicit conversions to and from int.  Probably it would need operators == and != and maybe < (or just <=> if we had C++20)